test_util/cluster_api.py

```diff
@@ -201,55 +202,50 @@ def __init__(self, dcos_uri, masters, public_masters, slaves, public_slaves,
 
         # URI must include scheme
         assert dcos_uri.startswith('http')
-        self.scheme = urlparse(dcos_uri).scheme
+        parse_result = urlparse(dcos_uri)
+        self.scheme = parse_result.scheme
+        self.dns_host = parse_result.netloc.split(':')[0]
 
         # Make URI never end with /
         self.dcos_uri = dcos_uri.rstrip('/')
 
-    @staticmethod
-    def _marathon_req_headers():
-        return {'Accept': 'application/json, text/plain, */*'}
+        # Generate request functions
+        for method in self.request_methods:
+            setattr(self, method, self.cluster_request(getattr(requests, method)))
 
-    def _suheader(self, disable_suauth):
-        if not disable_suauth and self.auth_enabled:
-            return self.web_auth_default_user.auth_header
-        return {}
-
-    def get(self, path="", params=None, disable_suauth=False, **kwargs):
-        hdrs = self._suheader(disable_suauth)
-        hdrs.update(kwargs.pop('headers', {}))
-        return requests.get(
-            self.dcos_uri + path, params=params, headers=hdrs, **kwargs)
-
-    def post(self, path="", payload=None, disable_suauth=False, **kwargs):
-        hdrs = self._suheader(disable_suauth)
-        hdrs.update(kwargs.pop('headers', {}))
-        if payload is None:
-            payload = {}
-        return requests.post(self.dcos_uri + path, json=payload, headers=hdrs)
-
-    def delete(self, path="", disable_suauth=False, **kwargs):
-        hdrs = self._suheader(disable_suauth)
-        hdrs.update(kwargs.pop('headers', {}))
-        return requests.delete(self.dcos_uri + path, headers=hdrs, **kwargs)
-
-    def head(self, path="", disable_suauth=False):
-        hdrs = self._suheader(disable_suauth)
-        return requests.head(self.dcos_uri + path, headers=hdrs)
-
-    def node_get(self, node, path="", params=None, disable_suauth=False, **kwargs):
-        """Execute a GET request against the adminrouter on node."""
-        hdrs = self._suheader(disable_suauth)
-        hdrs.update(kwargs.pop('headers', {}))
-
-        if node in self.masters:
-            port = self.adminrouter_master_port[self.scheme]
+    def cluster_request(self, request_fn):
```

Comment:
Can we make wrapping this just be binding parameters with partialmethod / partial rather than doing the decorator pattern? 

I find it a lot easier to not have the function in function and just read a couple of parameters being bound away.

```python
def cluster_request(self, request_method, path="", node=None, user=self.web_auth_default_user, port=None, **kwargs):
    ...
    requests.request(method, url, ...)
```

then we can do `self.get = functools.partialmethod(cluster_request, "get")`